### ISSUE TITLE
Could you please use 'rc' versions suffix instead of 'ea'?

### ISSUE THREAD

**Issue by remal** [immutables/immutables#1251]
Hello guys,

Recently two `ea` versions were released: `2.8.9-ea-0` and `2.8.9-ea-1`.

I use Renovate for automatic dependency updates and this tool treats `ea` versions as stable. I opened an issue about that (https://github.com/renovatebot/renovate/issues/7836), but I'd say that `ea` suffix is not that common and much more tool can treat such versions as stable.

I suggest using `rc` version suffix as much more stable.

**Comment by rarkins** [comment]
Per my understanding, the Maven versioning standard does not specify that having a suffix means a version is unstable. Only `alpha`, `beta` and `rc` and their shortened one-letter abbreviations are recognized by Maven. Therefore:
- `2.8.9-ea-1` is not an unstable version according to the Maven spec
- `2.8.9-ea-1` sorts higher than `2.8.8` so is a valid upgrade

Here's a link to the maven source which lists the well-known qualifiers: https://github.com/apache/maven/blob/3736be9c15122d65e6ef675557eb0f882f82b012/maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java#L44-L56

**Comment by rarkins** [comment]
A couple more points on versioning:
- You can see that alpha/beta/rc get special treatment while ea and others do not: https://mvnrepository.com/artifact/org.immutables/value
- The abnormally versioned `2.8.6.datatypes` sorts lower than `2.8.6` even though it was released after

![image](https://user-images.githubusercontent.com/6311784/100521693-6f345e80-31a5-11eb-8544-1b2df81c0a28.png)

**Comment by elucash** [comment]
ok to avoid `ea`, will use `rc`, `alpha`, `beta`. How then we should call completely unrelated special/experimental releases? would `rc-something` work?

**Comment by rarkins** [comment]
I've seen the react team using 0.0.0-alpha.* for completely experimental so that they don't sort higher than releases.

**Comment by elucash** [comment]
`0.0.0-alpha-experimental-N` Special/experimental releases
`0.0.0-alpha-N` Not feature complete
`0.0.0-beta-N` Feature complete, but unstable/buggy
`0.0.0-rc-N` Contains little to none known issues, release candidate
Reasonable?

**Comment by asereda-gs** [comment]
Agree. Just a small preference for `alphaN` (eg. `alpha1`) instead of `alpha-N` (eg. `alpha-1`).
Or `rc0` vs `rc-0`. 

Maven [should be able](https://maven.apache.org/ref/3.5.2/maven-artifact/apidocs/org/apache/maven/artifact/versioning/ComparableVersion.html) to transition between `alpha` and next digit.

**Comment by elucash** [comment]
ok
`0.0.0-alpha-experimentalN` (or `milestone-somethingN` )
`0.0.0-alphaN`
`0.0.0-betaN`
`0.0.0-rcN`

**Comment by xenoterracide** [comment]
+1 I know that I can configure gradle to exclude this stuff and am going to have to look it up, but, I'd rather not.
